src/components/AuthenticatedRoute/index.tsx

```diff
@@ -1,14 +1,21 @@
 import React from "react";
 import { Route, Redirect } from "react-router-dom";
+import { connect } from 'react-redux'
 
-export default ({ component: C, props: cProps, ...rest }) =>
+const AuthenticatedRoute = ({ component: C, cProps, ...rest }) =>
   <Route
     {...rest}
     render={props =>
       cProps.isAuthenticated
-        ? <C {...props} {...cProps} />
+        ? <C {...props}/>
         : <Redirect
             to={`/login?redirect=${props.location.pathname}${props.location
               .search}`}
           />}
   />;
+
+  const mapStateToProps = state => (
+    {cProps: state.authenticate}
```

Comment:
For this one, maybe you can declare used variable directly such as:
```
isAuthenticated: state.authenticate.isAuthenticated
```
The idea is we'll declare enough variables we'll use.